test/policies/user_policy_test.rb

```diff
@@ -8,6 +8,23 @@ def setup
     @user_as_department_manager = create :user, :with_profile, role: 'department_manager'
   end
 
+  test 'Destroy: superadmin can delete other user' do
+    assert_permit @user_as_superadmin, @user_as_social_worker, 'destroy?'
+    assert_permit @user_as_superadmin, @user_as_department_manager, 'destroy?'
+    assert_permit @user_as_other_superadmin, @user_as_social_worker, 'destroy?'
+    assert_permit @user_as_other_superadmin, @user_as_department_manager, 'destroy?'
+  end
+
+  test 'Destroy: superadmin cannot delete other superadmin' do
+    refute_permit @user_as_superadmin, @user_as_other_superadmin, 'destroy?'
+    refute_permit @user_as_other_superadmin, @user_as_superadmin, 'destroy?'
```

Comment:
I think that line is also not needed. The two superadmins are created in the same way, so I think only the first check is needed.